xalia/Win32/AccessibleProvider.cs

```diff
@@ -0,0 +1,305 @@
+        }
+
+        public UiDomValue EvaluateIdentifier(UiDomElement element, string identifier, HashSet<(UiDomElement, GudlExpression)> depends_on)
+        {
+            switch (identifier)
+            {
+                case "is_msaa_element":
+                    return UiDomBoolean.True;
+                case "msaa_role":
+                    depends_on.Add((Element, new IdentifierExpression("msaa_role")));
+                    return RoleAsValue;
+            }
+            return UiDomUndefined.Instance;
+        }
+
+        public UiDomValue EvaluateIdentifierLate(UiDomElement element, string identifier, HashSet<(UiDomElement, GudlExpression)> depends_on)
+        {
+            if (property_aliases.TryGetValue(identifier, out var aliased))
+            {
+                return Element.EvaluateIdentifier(aliased, Element.Root, depends_on);
+            }
+            if (msaa_name_to_role.TryGetValue(identifier, out var role))
+            {
+                depends_on.Add((Element, new IdentifierExpression("msaa_role")));
+                if (RoleKnown)
+                {
+                    return UiDomBoolean.FromBool(Role == role);
+                }
+            }
+            return UiDomUndefined.Instance;
+        }
+
+        public Task<(bool, int, int)> GetClickablePointAsync(UiDomElement element)
+        {
+            return Task.FromResult((false, 0, 0));
+        }
+
+        public string[] GetTrackedProperties()
+        {
+            return null;
+        }
+
+        public void NotifyElementRemoved(UiDomElement element)
+        {
+            Element = null;
+            IAccessible = null;
+        }
+
+        public void TrackedPropertyChanged(UiDomElement element, string name, UiDomValue new_value)
+        {
+        }
+
+        public bool UnwatchProperty(UiDomElement element, GudlExpression expression)
+        {
+            return false;
+        }
+
+        public bool WatchProperty(UiDomElement element, GudlExpression expression)
+        {
+            if (expression is IdentifierExpression id)
+            {
+                switch (id.Name)
+                {
+                    case "msaa_role":
+                        {
+                            if (!RoleKnown && !_fetchingRole)
+                            {
+                                _fetchingRole = true;
+                                Utils.RunTask(FetchRole());
+                            }
+                            break;
+                        }
+                }
+            }
+            return false;
+        }
+
+        private async Task FetchRole()
+        {
+            object role_obj;
+            try
+            {
+                role_obj = await Connection.CommandThread.OnBackgroundThread(() =>
+                {
+                    return IAccessible.accRole[ChildId];
+                }, RootHwnd.Tid + 1);
+            }

```

Comment:
It's a generic key. I figure if we have the tid then tid is probably better.

The +1 is a way of getting different queues. Since pids and tids are always divisible by 4, this makes it possible to put user actions (+0), queries (+1), and polls (+2) on different queues. I should probably document that somewhere.

Reply:
Yeah, I thought some more about it last night and doing it by TID when available makes sense. It would be nice to document the different queue values somewhere. :)

If we're going to use TID here instead of PID, would it make sense to change the variable names in `xalia/Win32/CommandThread.cs`?

Reply:
Changed the variable names and added some documentation to the file, does that work?

Reply:
Yeah it's better. The only thing that _maybe_ could improve it more would be an enum instead of a magic number, like `key + USER_REQ` or `key + POLL_REQ`. You could probably come up with better names for them if that's something you think is a good idea. Not a blocker or anything.